pyswarms/backend/topology/random.py

```diff
@@ -0,0 +1,193 @@
+        """Update the position matrix
+
+        This method updates the position matrix given the current position and
+        the velocity. If bounded, it waives updating the position.
+
+        Parameters
+        ----------
+        swarm : pyswarms.backend.swarms.Swarm
+            a Swarm instance
+        bounds : tuple of :code:`np.ndarray` or list (default is :code:`None`)
+            a tuple of size 2 where the first entry is the minimum bound while
+            the second entry is the maximum bound. Each array must be of shape
+            :code:`(dimensions,)`.
+
+        Returns
+        -------
+        numpy.ndarray
+            New position-matrix
+        """
+        return ops.compute_position(swarm, bounds)
+
+    def __compute_neighbors(self, swarm, k):
+        """Helper method to compute the adjacency matrix of the topology
+
+        This method computes the adjacency matrix of the topology using
+        the randomized algorithm proposed in [TSWJ2013]. The resulting
+        topology is a connected graph.
+
+        Parameters
+        ----------
+        swarm : pyswarms.backend.swarms.Swarm
+            a Swarm instance
+        k : int
+            number of neighbors to be considered. Must be a
+            positive integer less than :code:`n_particles-1`
+
+        Returns
+        -------
+        numpy.ndarray
+            Adjacency matrix of the topology
+        """
+
+        # Initialize an adjacency matrix as the identity matrix.
+        # This makes sure that every particle has itself as a
+        # neighbor
+        adj_matrix = np.identity(swarm.n_particles, dtype=int)
+
+        # Generate neighbor matrix. It has the indices of possible
+        # neighbors as its elements. The elements are generated randomly
+        neighbor_matrix = np.array(
+            [np.random.choice(
+                # Exclude i from the array
+                np.setdiff1d(
+                    np.arange(swarm.n_particles), np.array([i])
+                ), k, replace=False
+            ) for i in range(swarm.n_particles)])
+
+        # Set random elements to one using the neighbor matrix
+        adj_matrix[np.arange(swarm.n_particles).reshape(swarm.n_particles, 1), neighbor_matrix] = 1
+        adj_matrix[neighbor_matrix, np.arange(swarm.n_particles).reshape(swarm.n_particles, 1)] = 1
+
+        # Compute the distance matrix using Dijkstra's algorithm to
+        # later find unconnected pieces of the topology.
+        dist_matrix = dijkstra(adj_matrix, directed=False, return_predecessors=False, unweighted=True)
+
+        # Generate connected graph. If the graph isn't connected,
+        # it is possible that the PSO algorithm does not find the
+        # best position within the swarm. If the element (i,j) of the
+        # distance matrix is 0, that means that there is no path
+        # from node i to node j, thus there are still unconnected
+        # pieces of the graph (i.e. topology)
```

Comment:
Hi @whzup , one last thing regarding documentation. I'm not sure if you've seen my earlier comment but I suggested to put this long text on the docstring itself. You can just put a `# Generate connected graph` on the docstring. The idea is to make the docstring a bit longer while the in-line comments shorter (less clutter) and can be viewed by the developer using `help()` without resorting to look at the source code.